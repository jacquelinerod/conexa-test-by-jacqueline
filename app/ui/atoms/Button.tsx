import { MouseEventHandler, ReactNode } from "react";

interface IButton {
  disabled: boolean;
  onClick: MouseEventHandler<HTMLButtonElement>;
  className: string;
  text: string;
}

export default function Button({
  disabled,
  onClick,
  className,
  text,
}: IButton) {
  return (
    <button disabled={disabled} onClick={onClick} className={className}>
      {text}
    </button>
  );
}
