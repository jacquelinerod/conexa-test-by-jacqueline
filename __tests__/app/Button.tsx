import { render } from "@testing-library/react";
import Button from "@/app/ui/atoms/Button";

const mock = {
  disabled: true,
  onClick: jest.fn(),
  className: "myClass",
  text: "myText",
};

describe("Button Component", () => {
  it("should be disabled if the disabled property is true", () => {
    const button = (
      <Button
        disabled={true}
        onClick={mock.onClick}
        className={mock.className}
        text={mock.text}
      />
    );
    const { container } = render(button);
    expect(container).toMatchSnapshot();
  });

  it("should not be disabled if the disabled property is false", () => {
    const button = (
      <Button
        disabled={false}
        onClick={mock.onClick}
        className={mock.className}
        text={mock.text}
      />
    );
    const { container } = render(button);
    expect(container).toMatchSnapshot();
  });
});
