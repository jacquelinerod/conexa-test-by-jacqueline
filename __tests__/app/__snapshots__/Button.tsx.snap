// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should be disabled if the disabled property is true 1`] = `
<div>
  <button
    class="myClass"
    disabled=""
  >
    myText
  </button>
</div>
`;

exports[`Button Component should not be disabled if the disabled property is false 1`] = `
<div>
  <button
    class="myClass"
  >
    myText
  </button>
</div>
`;
